src/components/Checkbox/index.js

```diff
@@ -0,0 +1,53 @@
+import { Component } from 'pet-dex-utilities';
+import './index.scss';
+
+const events = ['change'];
+
+const html = `
+    <label class="checkbox-container">
+        <input data-select="checkbox" type="checkbox" value="">
+        <span data-select="checkbox-text" class="checkmark"></span>
+    </label>
+`;
+
+export default function Checkbox({ isChecked = false, text = '', isDisabled = false, value = '' }) {
+    Component.call(this, { html, events });
+
+    this.setChecked(isChecked);
+    this.setText(text);
+    this.setDisabled(isDisabled);
+    this.setValue(value);
+
+    const $checkbox = this.selected.get('checkbox');
+    const $checkboxText = this.selected.get('checkbox-text');
+
+    $checkbox.addEventListener('change', () => {
+        this.emit('change', $checkbox.checked);
+    });
+}
+
+Checkbox.prototype = Object.assign(Checkbox.prototype, Component.prototype, {
+    isChecked() {
+        return this.selected.get('checkbox').checked;
+    },
+
+    setChecked(isChecked = false) {
+        const $checkbox = this.selected.get('checkbox');
+        $checkbox.checked = isChecked;
+    },
```

Comment:
Sempre que mudamos um estado do componente, precisamos notificar a alteração.
```suggestion
    setChecked(isChecked = false) {
        const $checkbox = this.selected.get('checkbox');
        $checkbox.checked = isChecked;
         // seria legal se você emitir o valor antigo e o novo
        this.emit("change", isChecked);
    },
```